src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.HelperFunctions.cs

```diff
@@ -140,13 +141,16 @@ private static ExecutionPayloadV3 CreateBlockRequestV4(MergeTestBlockchain chain
             Block? block = blockRequestV4.TryGetBlock().Block;
 
             var beaconBlockRootHandler = new BeaconBlockRootHandler(chain.TxProcessor, chain.WorldStateManager.GlobalWorldState);
-            beaconBlockRootHandler.StoreBeaconRoot(block!, chain.SpecProvider.GetSpec(block!.Header), NullTxTracer.Instance);
+
+            IReleaseSpec spec = chain.SpecProvider.GetSpec(block!.Header);
+            var blkCtx = new BlockExecutionContext(block!.Header, spec);
+            beaconBlockRootHandler.StoreBeaconRoot(block!, in blkCtx, spec, NullTxTracer.Instance);
             IWorldState globalWorldState = chain.WorldStateManager.GlobalWorldState;
             Snapshot before = globalWorldState.TakeSnapshot();
             var blockHashStore = new BlockhashStore(chain.SpecProvider, globalWorldState);
             blockHashStore.ApplyBlockhashStateChanges(block!.Header);
 
-            chain.MainExecutionRequestsProcessor.ProcessExecutionRequests(block!, globalWorldState, [], chain.SpecProvider.GenesisSpec);
+            chain.MainExecutionRequestsProcessor.ProcessExecutionRequests(block!, globalWorldState, new BlockExecutionContext(block.Header, chain.SpecProvider.GenesisSpec), [], chain.SpecProvider.GenesisSpec);
```

Comment:
not sure if 0 basefee?

Reply:
Previously was usings header and spec passed in; not sure why would be 0 basefee? (System doesn't charge however)